gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/DatasetCleaner.java

```diff
@@ -88,33 +94,43 @@
 
   public DatasetCleaner(FileSystem fs, Properties props) throws IOException {
 
-    State state = new State(props);
+    Properties properties = new Properties();
+    properties.putAll(props);
+    // load dynamic configuration and add them to the job properties
+    Config propsAsConfig = ConfigUtils.propertiesToConfig(props);
+    DynamicConfigGenerator dynamicConfigGenerator =
+        DynamicConfigGeneratorFactory.createDynamicConfigGenerator(propsAsConfig);
+    Config dynamicConfig = dynamicConfigGenerator.generateDynamicConfig(propsAsConfig);
+    for (Map.Entry<String, ConfigValue> entry : dynamicConfig.entrySet()) {
+      properties.put(entry.getKey(), entry.getValue().unwrapped().toString());
```

Comment:
Does it work in this way ? It seems it will only get the top-level key and here it just convert the ConfigValue into a string which could itself be a hierarchical config.

Reply:
It was working well magically... But I change the code to avoid confusion